sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -184,14 +184,40 @@ def list_configuration_settings(
             for item in filtered_listed:
                 pass  # do something
         """
+
+    @overload
+    def list_configuration_settings(
+        self, *, snapshot_name: str, fields: Optional[List[str]] = None, **kwargs: Any
+    ) -> ItemPaged[ConfigurationSetting]:
+        """List the configuration settings stored under a snapshot in the configuration service, optionally filtered by
+        fields to present in return.
+
+        :keyword str snapshot_name: The snapshot name.
+        :keyword fields: Specify which fields to include in the results. Leave None to include all fields.
+        :type fields: list[str] or None
+        :return: An iterator of :class:`~azure.appconfiguration.ConfigurationSetting`
+        :rtype: ~azure.core.paging.ItemPaged[~azure.appconfiguration.ConfigurationSetting]
+        :raises: :class:`~azure.core.exceptions.HttpResponseError`
+        """
+
+    @distributed_trace
+    def list_configuration_settings(self, **kwargs) -> ItemPaged[ConfigurationSetting]:
         select = kwargs.pop("fields", None)
         if select:
             select = ["locked" if x == "read_only" else x for x in select]
+        snapshot_name = kwargs.pop("snapshot_name", None)
 
         try:
+            if snapshot_name is not None:
+                return self._impl.get_key_values(  # type: ignore
+                    snapshot=snapshot_name,
+                    select=select,
+                    cls=lambda objs: [ConfigurationSetting._from_generated(x) for x in objs],
+                    **kwargs
+                )
             return self._impl.get_key_values(  # type: ignore
-                label=label_filter,
-                key=key_filter,
+                key=kwargs.pop("key_filter", None),
```

Comment:
If user specify key_filter in both positional arg and kwarg, we should raise error to be consistent with old behavior?